homeassistant/components/netatmo/climate.py

```diff
@@ -317,18 +312,16 @@ def handle_event(self, event: dict) -> None:
 
             if (
                 data["event_type"] == EVENT_TYPE_CANCEL_SET_POINT
-                and self._id == room["id"]
+                and self._room.entity_id == room["id"]
             ):
                 self.async_update_callback()
                 self.async_write_ha_state()
                 return
 
     @property
-    def _data(self) -> pyatmo.AsyncHomeData:
+    def _data(self) -> pyatmo.AsyncClimate:
         """Return data for this entity."""
-        return cast(
-            pyatmo.AsyncHomeData, self.data_handler.data[self._data_classes[0]["name"]]
-        )
+        return cast(pyatmo.AsyncClimate, self._climate_state)
```

Comment:
No need to cast if you set type on assignment, and then I guess you can just drop `self._data` in favor of directly referencing `self._climate_state` ?